cpp/common/src/ext/stdc++.model.yml

```diff
@@ -0,0 +1,7 @@
+extensions:
+  - addsTo:
+      pack: codeql/common-cpp-coding-standards
+      extensible: throwingFunctionModel
+    data:
+      - ["std", "basic_string", "append", "std", "out_of_range"] 
```

Comment:
just to check, do we want to add anything else here?
cant think of a good systematic way to scrape these out of the cpp docs
but for example: [replace?](https://en.cppreference.com/w/cpp/string/basic_string/replace)
but there doesnt seem to be a top down list, [exception types say that they may be thrown by some members](https://en.cppreference.com/w/cpp/error/out_of_range) but not a full list of which...

Reply:
For this PR I wanted to limit it to just these cases.